### PR TITLE
Browser build

### PR DESCRIPTION
**Pull request by indolering**
Initial attempt at adding support for a browser build.

### CODE REVIEW

**Comment by indolering** [comment]
Fixes #137.

File: src/AbstractFontKit.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if(font === null){
+      throw new Error('Unknown font format');
+    } else {
+      return font;
+    }
+  }
+
+}
+
+export default AbstractFontKit;
```
Comment (indolering) [inline src/AbstractFontKit.js]: I didn't want Browserify to polyfill the fs module so I created this parent class.  If this looks good I can refactor index and base to use it.

File: src/browser.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      if(callback){
+        callback(error);
+      }
+
+      throw error;
+    });
+
+    return promise;
+  }
+}
```
Comment (indolering) [inline src/browser.js]: Returning a promise should be backwards compatible with the existing API....

**Comment by devongovett** [comment]
Why does the browserify build need to be in fontkit itself? Wouldn't you just browserify your entire project instead? For an example of using fontkit in the browser, check out [fontkit-demo](https://github.com/devongovett/fontkit-demo).

**Comment by talltyler** [comment]
Projects that use fontkit don't always use browserify to build.  The logic is every module should include the dependencies that it needs to run. Forcing every project that uses your code to also use your build system limits the number of people able to run your code.

**Comment by indolering** [comment]
> Why does the browserify build need to be in fontkit itself? Wouldn't you just browserify your entire project instead?

The build is designed to accommodate freestanding installs, so you can pull in fontkit via something like [cdnjs](https://cdnjs.com/) or [jsDelivr](https://www.jsdelivr.com/).

> Wouldn't you just browserify your entire project instead?

Not if you aren't using Browserify to begin with.  But this PR also makes it easier to do that via AbstractFontKit and the browser-specific FontKit classes.

> For an example of using fontkit in the browser, check out fontkit-demo.

Right, I'm trying to save people the hassle of reverse engineering your non-standard build system and writing their own loading function 😄 .

**Comment by jakubzitny** [comment]
This should definitely be merged. Or propose a solution that you'll be willing to merge @devongovett, please.

**Comment by scottrippey** [comment]
I've attempted to resurrect this PR, so I created #183 .
I didn't attempt to make the `AbstractFontKit`, so my PR still includes some of the `open` methods that would be broken.  But I guess that's an improvement that could be made separately.

**Comment by devongovett** [comment]
Handled in https://github.com/foliojs/fontkit/pull/276

**Pull request closed without merging**